### PR TITLE
Minor cleanups related to build-system

### PR DESCRIPTION
**Pull request by cgay**
Survives a 3-stage-bootstrap

### CODE REVIEW

File: sources/lib/build-system/build.dylan [hunk 1]
```diff
@@ -69,22 +69,24 @@ end macro;
...
-
+// Top level internal function that can be invoked by Dylan Clients
+// Arguments:
+// build-targets - A sequence of strings naming libraries to build.
+// directory - The directory in which to perform the build.
+// progress-callback - A function to call with build progress messages.
+//     Called with the message string and `warning?: #t` if the message
+//     is a warning.
+// build-script - Jam build script <file-locator>
+// compiler-back-end - #"llvm", #"c", #"harp", or #f.  #f means use HARP.
```
Comment (housel) [inline sources/lib/build-system/build.dylan]: ``#f`` doesn't mean use HARP, it just means the information is not provided or irrelevant. The actual back-end choice is made in ``make-project`` in the ``projects`` library.

File: sources/lib/build-system/build.dylan [hunk 2]
```diff
@@ -69,22 +69,24 @@ end macro;
 
 
 
-// Toplevel internal function that can be invoked by Dylan Clients
-
+// Top level internal function that can be invoked by Dylan Clients
+// Arguments:
+// build-targets - A sequence of strings naming libraries to build.
```
Comment (housel) [inline sources/lib/build-system/build.dylan]: The strings are Jam target names such as ``clean`` or ``clean-all``  or ``exe`` or ``dll``.

File: sources/lib/build-system/build.dylan (hunk 1 above)
Comment (cgay) [reply sources/lib/build-system/build.dylan]: Under what conditions is the back-end irrelevant?

File: sources/lib/build-system/build.dylan (hunk 2 above)
Comment (cgay) [reply sources/lib/build-system/build.dylan]: Thanks, that was rather brain dead.

**Pull request merged**